File: utils/consoleLogger.js
// utils/consoleLogger.js
module.exports = (page) => {
  page.on('console', msg => {
    const logEntry = {
      type: msg.type(),
      text: msg.text(),
      location: msg.location() ? {
        url: msg.location().url,
        line: msg.location().lineNumber,
        column: msg.location().columnNumber
      } : null,
      timestamp: new Date().toISOString()
    };

    // Affiche dans la console
    console.log(`[${logEntry.timestamp}] ${logEntry.type}: ${logEntry.text}`);
    if (logEntry.location) {
      console.log(`  @ ${logEntry.location.url}:${logEntry.location.line}:${logEntry.location.column}`);
    }

    // Optionnel : Sauvegarde dans un fichier ou une base de données
    require('fs').appendFileSync('console_logs.json', JSON.stringify(logEntry) + '\n');
  });
};

File: utils/console_logger.py
# utils/console_logger.py
import datetime

def setup_console_logger(page):
    def console_handler(msg):
        log_entry = {
            "type": msg.type,
            "text": msg.text,
            "location": {
                "url": msg.location.get("url", None),
                "line": msg.location.get("lineNumber", None),
                "column": msg.location.get("columnNumber", None)
            } if msg.location else None,
            "timestamp": datetime.datetime.now().isoformat()
        }
        print(f"[{log_entry['timestamp']}] {log_entry['type']}: {log_entry['text']}")
        if log_entry["location"]:
            print(f"  @ {log_entry['location']['url']}:{log_entry['location']['line']}:{log_entry['location']['column']}")

    page.on("console", console_handler)